## MEASURES OF CENTRAL TENDENCY
They only three for example: 
- Mean
- Median 
- Mode 


In [17]:
import pandas as pd 

student_data={
    "Name":["Jane","John","James","Joseph","Josephine","Andrew"],
    "Science":[89,98,79,79,69,56],
    "Maths":[91,86,95,69,89,66],
    "Gender":["Female","Male","Male","Male","Female","Male"]
}
students=pd.DataFrame(student_data)
students


,Name,Science,Maths,Gender
0,Jane,89,91,Female
1,John,98,86,Male
2,James,79,95,Male
3,Joseph,79,69,Male
4,Josephine,69,89,Female
5,Andrew,56,66,Male


In [37]:
students["Maths"].mode()[0]

np.int64(66)

EXCERCISE ONE 
1. Find the mean science mark and explain your result 
2. FIND THE MEDIAN OF THE SCIENCE MARKS AND LIST THE STUDENTS ATTACHED TO THAT MARK


Question 1

In [19]:
students["Science"].mean()

np.float64(78.33333333333333)

The mean for the science marks is 78.33 after adding the sum of the marks then dividing it by the number of the marks

Question 2

In [20]:
students["Science"].median()

np.float64(79.0)

In [28]:
students[students["Science"]==79][["Name","Science"]]

,Name,Science
2,James,79
3,Joseph,79


The median is the number in the middle of a given dataset which is 79.0

MEASURES OF DISPERSION
- Standard deviation
- Variance
- Range

In [29]:
## Standard Deviation 
students["Maths"].std()

np.float64(12.14358541233464)

In [31]:
## Variance 
students["Maths"].var()

np.float64(147.46666666666664)

In [32]:
## Range 
students["Maths"].max()-students["Maths"].min()

np.int64(29)

EXCERCISE TWO


1. FIND THE SPREAD OF THE SCIENCE MARKS USING VARIANCE 
2. WHAT IS THE RANGE OF THE SCIENCE MARKS AND COMMENT ON YOUR ANSWER 

QUESTION 1

In [21]:
students["Science"].var()

np.float64(217.46666666666664)

The spread of the science marks using variance is 217.466

QUESTION 2

In [22]:
students["Science"].max()-students["Science"].min()

np.int64(42)

The range of the science marks is 42 after subtracting the highest number from the lowest number 

MEASURES OF POSITION 
- PERCENTILES
- QUARTILES
- DECILES etc

In [23]:
## Quartiles
## lower quantile
lower_quantile=students["Maths"].quantile(0.25)
lower_quantile


np.float64(73.25)

In [24]:
## upper quantiles
upper_quantile = students["Maths"].quantile(0.75)
upper_quantile

np.float64(90.5)

PERCENTILES
- uses percentages 

In [25]:
import numpy as np


In [26]:
lower_quantile2=np.percentile(students["Maths"],25)
lower_quantile2

np.float64(73.25)

. find the 90th percentile for maths

In [27]:
ninty_th=np.percentile(students["Maths"],90)
ninty_th

np.float64(93.0)